src/Core.js

```diff
@@ -306,6 +306,10 @@ function getLocation(url){
     }
     // #endif
     var m = url.toLowerCase().match(reURI);
+    //if the origin is non standard url, such as chrome extensions
+    if(!m) {
```

Comment:
I think it is better than throwing an error, for several reasons: 
1. I don't think an error should be thrown at all, as this is not an error from the code, but rather a misuse of external code.
2.  Throwing an error will "spam" the console.
3. There's nothing you can do to avoid this error, so it will always be thrown.